### ISSUE TITLE
Add support of BLAKE3 hashes

### ISSUE THREAD

**Issue by ghost** [gtkhash/gtkhash#84]
Well in title is there anyway to do implement into gtkhash?

**Comment by ZarusMods** [comment]
And also how can i make batch file that launches .exe and then launch hashes

**Comment by tristanheaven** [comment]
Yes it's possible, but it would be easier if BLAKE3-team created a library with a C API - like BLAKE2/libb2.

**Comment by tristanheaven** [comment]
For the batch file, you can add file paths as command arguments and hashing should start automatically.

**Comment by ZarusMods** [comment]
NICE,thanks for help.

**Comment by ZarusMods** [comment]
> 
> 
> For the batch file, you can add file paths as command arguments and hashing should start automatically.

here is my batch file sample
`@echo off
cd Verify
Verify.exe -C:\Game\Verify\hashes`

**Comment by oconnor663** [comment]
> Yes it's possible, but it would be easier if BLAKE3-team created a library with a C API - like BLAKE2/libb2.

You can find [the official C implementation here](https://github.com/BLAKE3-team/BLAKE3/tree/master/c). (Maybe we should make that more discoverable? The top level readme links to it, but it's pretty far down.) Single-threaded performance is equivalent to the Rust implementation, because they use the same assembly files under the hood. Multi-threading isn't supported yet though.

**Comment by tristanheaven** [comment]
@oconnor663  It's fine, I found it there. The issue is that the C implementation doesn't have a build system and pkg-config file which would make it suitable for packaging in a distribution. Unless it gets included in another library (e.g. openssl), every application will need to have the same architecture/OS-specific build handling, or only use the portable C version.

**Comment by oconnor663** [comment]
Yeah that's a problem. My worry has been that I don't know how to provide a single build system that's useful to different callers, for example both callers using autotools and callers using Visual Studio. I've been meaning to try something with CMake, but I don't have enough experience to know in advance whether that's going to be useful to everyone. Do you know if that's a good approach? Anyway, apologies for derailing this thread :)
